benefits/core/context_processors.py

```diff
@@ -11,6 +12,38 @@ def analytics(request):
     return {"analytics": {"api_key": settings.ANALYTICS_KEY, "uid": session.uid(request), "did": session.did(request)}}
 
 
+def authentication(request):
+    """Context processor adds authentication information to request context."""
+    verifier = session.verifier(request)
+
+    if verifier:
+        return {
+            "authentication": {
+                "required": verifier.requires_authentication,
+                "logged_in": session.logged_in(request),
+                "sign_out_route": reverse("oauth:logout"),
+            }
+        }
+    else:
+        return {}
+
+
+def auth_provider(request):
+    """Context processor adds auth_provider information to request context."""
+    verifier = session.verifier(request)
+
+    if verifier and verifier.requires_authentication:
+        auth_provider = verifier.auth_provider
+        return {
+            "auth_provider": {
+                "sign_in_button_label": auth_provider.sign_in_button_label,
+                "sign_out_button_label": auth_provider.sign_out_button_label,
```

Comment:
I feel like these two properties can be combined with the `authentication` context no?

Instead of immediately returning the dict, build it first so you can optionally add the extra keys:

```python
if verifier:
    data = {
        "required": verifier.requires_authentication,
        "logged_in": session.logged_in(request),
        "sign_out_route": reverse("oauth:logout"),
    }
    
    if verifier.requires_authentication:
        # add the other fields
    
    return { "authentication": data }
```

Reply:
Done in 1a12137